internal/controller/streams.go

```diff
@@ -196,12 +193,38 @@ func (sm *streamManager) StreamResources(ctx context.Context, logger *zap.Logger
 		logger.Error("Failed to send cluster metadata", zap.Error(err))
 		return err
 	}
+
+	resourceVersions := make(map[string]string)
 	for resource, apiGroup := range resourceAPIGroupMap {
```

Comment:
In each loop, you should check `ctx.Done()` for every iteration.
```suggestion
	for resource, apiGroup := range resourceAPIGroupMap {
		select {
		case <-ctx.Done():
			return ctx.Err()
		default:
		}
```